### PR TITLE
PoC implementation

### PR DESCRIPTION
**Pull request by kfswain**
Implementation of PoC shared in wg-serving.

Making the code public so that other contributors can test our implementation, reproduce numbers and check assumptions.

### CODE REVIEW

File: examples/poc/manifests/gateway.yaml [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+                          - name: backend
+                            domains: ["*"]
+                            routes:
+                              - match:
+                                  prefix: "/"
+                                route:  
+                                  cluster: original_destination_cluster
+                                  timeout: 1000s  # Increase route timeout
+                      http_filters:
+                        - name: envoy.filters.http.ext_proc
```
[the remark this answers is not in the document]
Comment (smarterclayton) [reply examples/poc/manifests/gateway.yaml:53]: How widely is the API supported?  If it's more widely supported than the bootstrap config (or dramatically easier for most who would want to run the demo), then I agree we should follow up with that quickly.  Or present it as an alternative.

**Comment by smarterclayton** [comment]
If this runs, I would suggest we merge and move ahead and take additional feedback as comments to the merged PR or new PRs / issues.  The value of the PoC is people trying it, not reviewing it.

Also, I'd suggest we follow Kube conventions for commits, which are:

> One line below commit title length - summarize the issue (70? chars)
> any other description in a separate paragraphs

Squash commits and rebase down to minimal diffs (you can have multiple commits if each is logically distinct, like adding a test that fails and then fixing the issue, or delivering a fix before your main feature changes).

**Comment by kfswain** [comment]
> Squash commits and rebase down to minimal diffs 

Ack. Can we do this at merge? Or is it typically before?

**Comment by smarterclayton** [comment]
> At merge or before

Generally recommend everyone squash before they get approval, with a good comment and description that a future version of yourself and others can interpret to understand why the code was included.

**Review by terrytangyuan — COMMENTED** [review]
> If this runs, I would suggest we merge and move ahead and take additional feedback as comments to the merged PR or new PRs / issues. The value of the PoC is people trying it, not reviewing it.

I agree. It would be easier to test out once the PR is merged. This project is in early phase without releases yet so people should understand the risk.

**Comment by kfswain** [comment]
> > At merge or before
> 
> Generally recommend everyone squash before they get approval, with a good comment and description that a future version of yourself and others can interpret to understand why the code was included.

Done

File: examples/poc/README.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+## Requirements
+- A vLLM based deployment (using the custom image provided below), with LoRA Adapters
+- Kubernetes cluster
+- Envoy Gateway v1.1 installed on your cluster: https://gateway.envoyproxy.io/v1.1/tasks/quickstart/
+- `kubectl` command-line tool
+- Go (for local development)
+
+## vLLM
+***This PoC uses a modified vLLM fork, the public image of the fork is here: `ghcr.io/tomatillo-and-multiverse/vllm:demo`***
```
Comment (terrytangyuan) [inline examples/poc/README.md:13]: Is there a reason to use a modified fork?

File: examples/poc/README.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+
+## Overview
+
+This project contains the necessary configurations and code to set up and deploy a service using Kubernetes, Envoy, and Go. The service involves routing based on the model specified (using Open AI API format), collecting metrics, and ensuring efficient load balancing.
+
+![alt text](https://github.com/tomatillo-and-multiverse/lora-inference-gateway/blob/final-poc/envoy-gateway-bootstrap.png)
```
Comment (terrytangyuan) [inline examples/poc/README.md:20]: This is 404

File: examples/poc/README.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+3. Make your changes.
+4. Open a pull request.
+
+## License
+
+This project is licensed under the MIT License.
+
+---
+
+Feel free to customize this README to better fit your specific project details and requirements.
```
Comment (terrytangyuan) [inline examples/poc/README.md:63]: Remove this?

File: examples/poc/README.md (hunk 2 above)
Comment (smarterclayton) [reply examples/poc/README.md:13]: Document why the fork.

Comment (smarterclayton) [reply examples/poc/README.md:13]: And specifically link to the commits/commit-range/branch/PR that added them.

Comment (kfswain) [reply examples/poc/README.md:13]: Done. I linked to the fork, there is not currently a PR for the fork

**Pull request merged**